app/models/service_template_provision_task.rb

```diff
@@ -9,6 +9,10 @@ def self.base_model
     ServiceTemplateProvisionTask
   end
 
+  def my_zone
+    dialog_zone || source.my_zone || MiqServer.my_zone
```

Comment:
Previously we were checking if `source` exposed `my_zone` which was added in https://github.com/ManageIQ/manageiq/commit/f928e88545292f33189cf1e11f7acb092ba67a3f#diff-3d5779506719c23b56989155988a6ef9R143

Is that no longer a concern?

Reply:
```my_zone``` is added to ServiceTemplate in this PR. So I think all ```source``` here would respond to ```my_zone```.